worker/src/main/java/gov/cms/ab2d/worker/bfdhealthcheck/BFDHealthCheck.java

```diff
@@ -0,0 +1,64 @@
+package gov.cms.ab2d.worker.bfdhealthcheck;
+
+import gov.cms.ab2d.bfd.client.BFDClient;
+import gov.cms.ab2d.common.dto.PropertiesDTO;
+import gov.cms.ab2d.common.service.PropertiesService;
+import lombok.extern.slf4j.Slf4j;
+import org.hl7.fhir.dstu3.model.CapabilityStatement;
+import org.hl7.fhir.dstu3.model.Enumerations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+import static gov.cms.ab2d.common.util.Constants.MAINTENANCE_MODE;
+
+@Component
+@Slf4j
+class BFDHealthCheck {
+
+    @Autowired
+    private PropertiesService propertiesService;
+
+    @Autowired
+    private BFDClient bfdClient;
+
+    @Value("${bfd.health.check.consecutive.successes}")
+    private int consecutiveSuccessesToBringUp;
+
+    @Value("${bfd.health.check.consecutive.failures}")
+    private int consecutiveFailuresToTakeDown;
+
+    // Nothing else should be calling this component except for the scheduler, so keep
+    // state here
+    private int consecutiveSuccesses = 0;
+
+    private int consecutiveFailures = 0;
+
+    void checkBFDHealth() {
+
+        CapabilityStatement capabilityStatement = bfdClient.capabilityStatement();
+        if (capabilityStatement == null || capabilityStatement.getStatus() != Enumerations.PublicationStatus.ACTIVE) {
+            consecutiveFailures++;
+            consecutiveSuccesses = 0;
+        } else {
+            consecutiveSuccesses++;
```

Comment:
The problem with this logic is that it will keep counting the successes even when *not* in maintenance made and then constantly updating the property to `false` even though it is already `false`.